### PR TITLE
OEL-2766: Allow users to subscribe to events.

### PR DESCRIPTION
**Pull request by brummbar**

### CODE REVIEW

File: modules/oe_showcase_subscriptions/oe_showcase_subscriptions.info.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: OpenEuropa Showcase Subscriptions
+type: module
+description: Provides user subscriptions to content types.
+core_version_requirement: ^9.4 || ^10
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/oe_showcase_subscriptions.info.yml]: We can drop 9 support

File: modules/oe_showcase_subscriptions/oe_showcase_subscriptions.module [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @file
+ * OE Showcase Subscriptions module.
+ */
+
+declare(strict_types = 1);
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/oe_showcase_subscriptions.module]:
```suggestion
declare(strict_types=1);
```

File: modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    $state->delete('message_digest:weekly_last_run');
+    \Drupal::time()->resetTime();
+  }
+
+  /**
+   * Tests the event subscription and notifications.
+   */
+  public function testEventNotifications(): void {
+    $event = $this->createEventNode();
+    $this->markEntityForCleanup($event);
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php]: Not needed, create function already marks it

File: composer.json [hunk 4]
```diff
@@ -50,15 +53,17 @@
         "openeuropa/oe_oembed": "^0.7.0",
         "openeuropa/oe_paragraphs": "^1.21",
         "openeuropa/oe_starter_content": "^1.2.0",
+        "openeuropa/oe_subscriptions": "1.x-dev",
         "openeuropa/oe_webtools": "^1.23",
-        "openeuropa/oe_whitelabel": "^1.3.0"
+        "openeuropa/oe_whitelabel": "0.2767.202303041517"
```
Comment (AaronGilMartinez) [inline composer.json:58]: @todo to be changed

File: modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+      [
+        'url' => $event->toUrl()->setAbsolute()->toString(),
+        'text' => 'Event update 2',
+      ],
+    ]);
+
+    // Move the time forward one day.
+    \Drupal::time()->setTime(strtotime('+1 day +1 hour', \Drupal::time()->getCurrentTime()));
+    $this->cronRun();
+    $this->waitUntilMailsAreCollected(1);
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php:138]: Maybe comment why we have waits?

File: modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+
+    $this->drupalLogin($editor);
+    // Update the second event.
+    $this->drupalGet($event_two->toUrl('edit-form'));
+    $assert_session->fieldExists('Title')->setValue('Second event - update 1');
+    $assert_session->buttonExists('Save')->press();
+    $assert_session->pageTextContains('Event Second event - update 1 has been updated.');
+    // Update also the first event.
+    $this->drupalGet($event->toUrl('edit-form'));
+    $assert_session->fieldExists('Title')->setValue('Event update 4');
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php:210]: Nitpick: Can we use same nomemclature for both events? 'Ordinal event - update n'

File: modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+        ':content' => $content,
+      ]);
+      $assert_session->elementExists('xpath', $xpath);
+    }
+  }
+
+  /**
+   * Retrieves the URIs of links existing in a mail.
+   *
+   * @param \Drupal\symfony_mailer\Email $mail
```
Comment (AaronGilMartinez) [inline modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php:287]: readMail() returns \Symfony\Component\Mime\Email
Should we change this?

**Review by AaronGilMartinez — CHANGES_REQUESTED** [review]

File: modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php (hunk 7 above)
Comment (brummbar) [reply modules/oe_showcase_subscriptions/tests/src/ExistingSite/EventSubscriptionTest.php:287]: It says it returns that but actually collected emails are of type \Drupal\symfony_mailer\Email.

**Review by AaronGilMartinez — APPROVED** [review]

**Pull request merged**